dockerfile-generator/dockerfile-gen-services/src/main/java/io/hyscale/dockerfile/gen/services/exception/DockerfileErrorCodes.java

```diff
@@ -15,28 +15,40 @@
  */
 package io.hyscale.dockerfile.gen.services.exception;
 
-import io.hyscale.commons.exception.HyscaleErrorCode;
+import io.hyscale.commons.exception.HyscaleError;
+import io.hyscale.commons.exception.HyscaleErrorGroup;
 
-public enum DockerfileErrorCodes implements HyscaleErrorCode {
-    FAILED_TO_GENERATE_DOCKERFILE("Failed to generate Dockerfile"),
+public enum DockerfileErrorCodes implements HyscaleError {
+    FAILED_TO_GENERATE_DOCKERFILE("Failed to generate Dockerfile",HyscaleErrorGroup.DOCKER_FILE_GENERATION),
     FAILED_TO_COPY_ARTIFACT("Failed to copy artifacts"),
     FAILED_TO_CREATE_SCRIPT("Failed to create script"),
     FAILED_TO_PERSIST_DOCKERFILE("Failed to persist Dockerfile"),
-    BUILD_SPEC_REQUIRED("BuildSpec Required"),
-    DOCKERFILE_OR_BUILDSPEC_REQUIRED("Either Dockerfile or BuildSpec is required to build an image"),
-    FAILED_TO_PROCESS_DOCKERFILE_GENERATION("Failed to process Dockerfile Generation"),
+    BUILD_SPEC_REQUIRED("BuildSpec Required",HyscaleErrorGroup. DOCKER_FILE_GENERATION),
+    DOCKERFILE_OR_BUILDSPEC_REQUIRED("Either Dockerfile or BuildSpec is required to build an image",HyscaleErrorGroup.DOCKER_FILE_GENERATION),
```

Comment:
This is an image building error group, can you verify this case once again ?

Reply:
I assigned error group codes to the errors based on the below sheet. Can you please verify it.

https://docs.google.com/document/d/19rU4jD2dXYj9S_vn6b0jLJ_vivXOrgFUMX7fNRDNBEk/edit

BUILD_SPEC_REQUIRED and DOCKERFILE_OR_BUILDSPEC_REQUIRED are being used in the docker-file-gen service calls. Please find below image for reference.

![image](https://user-images.githubusercontent.com/66588430/84874801-bf5daf80-b0a2-11ea-8d19-ffef4f6a07ab.png)

Reply:
As per discussion error message for DOCKERFILE_OR_BUILDSPEC_REQUIRED is changed to Expected either Dockerfile or BuildSpec. Cannot accept both to define image spec